packages/core/src/llm/LLM.ts

```diff
@@ -129,7 +129,7 @@ export class OpenAI implements LLM {
   hasStreaming: boolean = true;
 
   // Per completion OpenAI params
-  model: keyof typeof ALL_AVAILABLE_OPENAI_MODELS;
+  model: keyof typeof ALL_AVAILABLE_OPENAI_MODELS | string;
```

Comment:
what's the advantage of allowing string here? The disadvantage that I see is that it's easy to type a wrong model name, e.g. `"gpt-3.5-turbo-1107"` and then our code won't work

Reply:
But it will be easy to extend the class since some models, like together ai use the same RESTful API with open ai.

Reply:
but aren't you already overriding model to `string` in together ai?

Reply:
what i don't like here is that we should have a type alias for `keyof typeof ALL_AVAILABLE_OPENAI_MODELS` so we can use that in `model`